tests/test_auth_backends.py

```diff
@@ -162,3 +162,12 @@ def test_middleware_response_header(self):
         response = m(request)
         self.assertIn("Vary", response)
         self.assertIn("Authorization", response["Vary"])
+
+    def test_get_default_redirect_uri_empty_redirect_uri_raise_exception(self):
+        self.application.authorization_grant_type = Application.GRANT_AUTHORIZATION_CODE
```

Comment:
See the tox log. It looks like this code is failing due to an undefined variable. I suggest you run tox locally before pushing the next commit.

```
=================================== FAILURES ===================================
  _ TestOAuth2Middleware.test_get_default_redirect_uri_empty_redirect_uri_raise_exception _
  
  self = <tests.test_auth_backends.TestOAuth2Middleware testMethod=test_get_default_redirect_uri_empty_redirect_uri_raise_exception>
  
      def test_get_default_redirect_uri_empty_redirect_uri_raise_exception(self):
  >       self.application.authorization_grant_type = Application.GRANT_AUTHORIZATION_CODE
  E       NameError: name 'Application' is not defined
```